Backend/TimeTracker.Api/Controllers/ProjectController.cs

```diff
@@ -59,24 +59,37 @@ public async Task<GenericResponseDTO<Project>> GetProjectById(int id)
         /// </summary>
         /// <returns>A list of all the projects the user is the teacher or is a student of</returns>
         [Authorize]
-        [HttpGet]
-        public async Task<GenericResponseDTO<List<Project>>> GetProjectsByUser()
+        [HttpGet]
+        [Route("Bool")]
+        public async Task<GenericResponseDTO<List<Project>>> GetProjectsByUser(string activeOnly)
         {
-            var currentUserId = authHelper.GetCurrentUserId(User);
-
-            // Get all the projects we teach or are a student of
-            var projects = await database.Projects
-                .AsNoTracking()
-                .Where(x => x.Teacher.Id == currentUserId || x.Students.Any(x => x.Id == currentUserId))
-                .Include(x => x.Teacher)
-                .Include(x => x.Students)
-                .Include(x => x.Tags)
-                .ToListAsync();
-
-            return new GenericResponseDTO<List<Project>>() 
-            { 
-                Data = projects,
-                Success = true
+            var currentUserId = authHelper.GetCurrentUserId(User);
+
+            // All projects we teach or are a student of
+            var projects = await database.Projects
+              .AsNoTracking()
+              .Where(x => (x.Teacher.Id == currentUserId || x.Students.Any(x => x.Id == currentUserId)))
+              .Include(x => x.Teacher)
+              .Include(x => x.Students)
+              .Include(x => x.Tags)
+              .ToListAsync();
+
+            // Only (active) projects we teach or are a student of
+            if (activeOnly == "true")
+            {
+                projects = await database.Projects
+                  .AsNoTracking()
+                  .Where(x => (x.Teacher.Id == currentUserId || x.Students.Any(x => x.Id == currentUserId)) && x.ArchivedDate == null)
+                  .Include(x => x.Teacher)
+                  .Include(x => x.Students)
+                  .Include(x => x.Tags)
+                  .ToListAsync();
+            }

```

Comment:
Looks like 2 API calls are being made here if its true, the top one should probably be in the else statement

Reply:
Ohhh good point thanks!

Reply:
Corrected if statements.